File: LMS_CMS_Backend/LMS_CMS_DAL/Models/Domains/LMS/LessonActivity.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LMS_CMS_DAL.Models.Domains.LMS
{
    public class LessonActivity : AuditableEntity
    {
        [Key]
        public long ID { get; set; }
        [Required(ErrorMessage = "English Title is required")]
        [StringLength(100, ErrorMessage = "English Title cannot be longer than 100 characters.")]
        public string EnglishTitle { get; set; }
        [Required(ErrorMessage = "Arabic Title is required")]
        [StringLength(100, ErrorMessage = "Arabic Title cannot be longer than 100 characters.")]
        public string ArabicTitle { get; set; }
        public string? AttachmentLink { get; set; }
        public int Order { get; set; }
        public string? Details { get; set; }

        [ForeignKey("Lesson")]
        public long LessonID { get; set; }
        public Lesson Lesson { get; set; }

        [ForeignKey("LessonActivityType")]
        public long LessonActivityTypeID { get; set; }
        public LessonActivityType LessonActivityType { get; set; }
    }
}

File: LMS_CMS_Frontend/src/app/Models/LMS/lesson-activity.ts
export class LessonActivity {
    constructor(
        public id: number = 0,
        public englishTitle : string = '',
        public arabicTitle  : string = '', 
        public details: string = '',
        public order: number|null = null,
        public lessonID: number = 0,
        public lessonEnglishTitle : string = '',
        public lessonArabicTitle  : string = '',
        public lessonActivityTypeID: number = 0,
        public lessonActivityTypeEnglishName : string = '',
        public lessonActivityTypeArabicName  : string = '',
        public insertedByUserId: number = 0,
        public attachmentFile : File|null = null,
        public attachmentLink : string = ""
    ){}
}  